packages/app/app/studio/[organization]/library/[session]/components/UploadToYoutubeButton.tsx

```diff
@@ -0,0 +1,130 @@
+'use client'
+import { Button } from '@/components/ui/button'
+import {
+  Dialog,
+  DialogContent,
+  DialogTrigger,
+} from '@/components/ui/dialog'
+import { IExtendedOrganization } from '@/lib/types'
+import Link from 'next/link'
+
+import React, { useState } from 'react'
+
+const NoYoutubeChannelModal = ({
+  hasChannel,
+}: {
+  hasChannel?: string
+}) => {
+  const hasNoChannel = hasChannel === 'noChannel'
+  const [open, setOpen] = useState(hasNoChannel ?? false)
+  return (
+    <Dialog open={open} onOpenChange={setOpen}>
+      <DialogContent className="px-4">
+        <h4 className="text-xl font-bold">
+          Create Youtube Channel Before Continuing
+        </h4>
+        <p>
+          To upload a video to YouTube, YouTube requires you create an
+          account. goto{' '}
+          <Link
+            className="text-destructive"
+            target="_blank"
+            rel="noopener"
+            href="https://www.youtube.com">
+            www.youtube.com
+          </Link>{' '}
+          to create one, return to StreamEth to connect again.
+        </p>
+      </DialogContent>
+    </Dialog>
+  )
+}
+
+const UploadToYoutubeButton = ({
+  organization,
+  organizationSlug,
+  sessionId,
+  hasChannel,
+}: {
+  organization: IExtendedOrganization | null
+  organizationSlug: string
+  sessionId: string
+  hasChannel?: string
+}) => {
+  const [selectedDestination, setSelectedDestination] =
+    useState<string>()
+  const clientId = process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID || ''
+  const redirectUri =
+    process.env.NEXT_PUBLIC_GOOGLE_REDIRECT_URI || ''
+
+  const handleYoutubeClick = () => {
+    const state = encodeURIComponent(
+      JSON.stringify({
+        redirectUrl: `/studio/${organizationSlug}/library/${sessionId}`,
+        organizationId: organization?._id,
+      })
+    )
+    // Encode the redirect URL
+    const authUrl = `https://accounts.google.com/o/oauth2/auth?client_id=${clientId}&redirect_uri=${redirectUri}&response_type=code&access_type=offline&scope=https://www.googleapis.com/auth/youtube.upload https://www.googleapis.com/auth/youtube.readonly https://www.googleapis.com/auth/youtube https://www.googleapis.com/auth/userinfo.profile https://www.googleapis.com/auth/userinfo.email&state=${state}`
+    window.location.href = authUrl
+  }
+  const hasSocials = organization?.socials?.length
+    ? organization?.socials?.length > 0
+    : false
+
+  const handlePublish = () => {}
```

Comment:
I guess this is for part two of the PR right?

Reply:
Yes, awaiting @Eric-Vondee feedback on the best form of payload to send from the frontend to the publish api